src/libstore/build/worker.cc

```diff
@@ -79,27 +89,13 @@ std::shared_ptr<DerivationGoal> Worker::makeBasicDerivationGoal(const StorePath
 
 std::shared_ptr<PathSubstitutionGoal> Worker::makePathSubstitutionGoal(const StorePath & path, RepairFlag repair, std::optional<ContentAddress> ca)
 {
-    std::weak_ptr<PathSubstitutionGoal> & goal_weak = substitutionGoals[path];
-    auto goal = goal_weak.lock(); // FIXME
```

Comment:
There was a FIXME here. I don't know what it was referring to, but whatever it was I don't think it has been addressed in the new function (which dropped the FIXME)?

Reply:
It has indeed not been fixed --- the new function should have the exact same behavior, but without the "meaningless" mutation of the local `goal` variable, but I have no idea what the `FIXME` means.

Reply:
The FIXME comes from bbe97dff8b3054d96e758f486f9ce3fa09e64de3 but if the issue was pointlessly copying the path, I think that was a preexisting issue?

Can't say I really care much about that issue. C++ makes not doing that rather annoying

Reply:
Yeah that doesn't sound like a big deal.